src/Constraint/RequestConstraintFactory.php

```diff
@@ -24,9 +25,20 @@ public function createConstraint(ValidationRules $validationRules): RequestConst
     {
         $options = [];
         foreach ($validationRules->getDefinitions() as $key => $definitions) {
-            $options[$key] = $this->factory->fromRuleDefinitions($definitions);
+            $options[$key] = $this->factory->fromRuleDefinitions($definitions, $validationRules->getAllowExtraFields());
         }
 
+        /**
+         * @var array{
```

Comment:
Would be nice to alias this? Definition kinda getting out of hand ><

https://phpstan.org/writing-php-code/phpdoc-types#local-type-aliases

Reply:
On second thought, it wouldve been ideal if we could declare a more global type. The current "global type" is declared in phpstan configuration file, which probably doesn't get read when you use this library. Can leave this as-is